src/concrete_intersection.jl

```diff
@@ -364,23 +364,33 @@ A new same polytope in H-representation with just one more constraint.
 """
 function intersection(P1::HPoly{N},
                       P2::HPoly{N};
-                      backend=default_polyhedra_backend(P1, N),
-                      prunefunc=removehredundancy!) where {N<:Real}
+                      backend=nothing,
+                      prunefunc=remove_redundant_constraints!) where {N<:Real}
     if typeof(P1) == typeof(P2)
         HPOLY = typeof(P1)
     else
         # one of them must be a polytope, so the intersection will be bounded
         HPOLY = HPolytope{N}
     end
+
     # concatenate the linear constraints
     Q = HPOLY([constraints_list(P1);
                constraints_list(P2)])
 
     # remove redundant constraints
-    if prunefunc == removehredundancy!
+    if prunefunc == remove_redundant_constraints!
+        if backend == nothing
+            backend = GLPKSolverLP()
```

Comment:
i thought about this but i didn't come up with a simpler way.

basically, the `prunefunc` specifies the algorithm to be used, and the backend is an option to it, but each algorithm needs a different default backend.

Reply:
Okay, I see that it is not so simple. But since you do a case split based on the function name, should we instead use clearer names, like `"polyhedra"` and `"default"`? You anyway do not use `prunefunc` inside the cases.